src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * A class that encapsulates the loading tasks and saving of tasks
+ * to file.
+ */
+public class Storage {
+    /**
+     * Class constructor that initialises the path where we will
+     * save the data in.
+     *
+     * @throws DukeException If we are unable to create the 'data'
+     *                       directory.
+     */
```

Comment:
I think it is really important to explain why exceptions were thrown. Nice job here!